sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -149,23 +148,23 @@ def set_json_body(json):
         headers.update({"Content-Length": str(len(body))})
     return headers, body
 
-def lookup_encoding(encoding):
-    # type: (str) -> bool
+def lookup_encoding(encoding: str) -> bool:
     # including check for whether encoding is known taken from httpx
     try:
         codecs.lookup(encoding)
         return True
     except LookupError:
         return False
 
-def get_charset_encoding(response):
-    # type: (...) -> Optional[str]
+def get_charset_encoding(response) -> Optional[str]:
     content_type = response.headers.get("Content-Type")
 
     if not content_type:
         return None
-    _, params = cgi.parse_header(content_type)
-    encoding = params.get('charset') # -> utf-8
+    # https://peps.python.org/pep-0594/#cgi
+    m = email.message.Message()
```

Comment:
The PEP suggests to use `email.message.Message` but per the [docs](https://docs.python.org/3/library/email.compat32-message.html) do we want to use EmailMessage ? Im unsure if Compat32 applies to us, hence the question

Reply:
I followed to the letter the PEP example :)
![image](https://user-images.githubusercontent.com/1050156/209890322-f8efbf89-2348-449b-9929-64c1510168e9.png)